pkg/deviceplugin/nic.go

```diff
@@ -3,69 +3,97 @@
 package deviceplugin
 
 import (
+	"errors"
 	"os"
-	"os/exec"
-	"regexp"
+	"path"
+	"strconv"
 	"strings"
 
 	"github.com/golang/glog"
 )
 
-// Takes the output from lshw and returns the device name for each solarflare
-// device.
-func parseOutput(output string) []string {
-	// "lshw -short -class network" sample output:
-	// H/W path            Device     Class          Description
-	// =========================================================
-	// /0/100/1b/0         enp2s0f0   network        XtremeScale SFC9250 10/25/40/50/100G Ethernet Controller
-	// /0/100/1b/0.1       enp2s0f1   network        XtremeScale SFC9250 10/25/40/50/100G Ethernet Controller
-	// /0/100/1c.1/0       eno1       network        NetXtreme BCM5720 Gigabit Ethernet PCIe
-	// /0/100/1c.1/0.1     eno2       network        NetXtreme BCM5720 Gigabit Ethernet PCIe
+const (
+	sysClassNetPath  = "/sys/class/net/"
+	solarflareVendor = "0x1924"
+)
+
+func isSFCNic(devicePath string) bool {
+	deviceDir, err := os.Stat(path.Join(devicePath, "device"))
+	if errors.Is(err, os.ErrNotExist) {
+		// Not a physical device, so won't have the "vendor" file
+		return false
+	} else if err != nil {
+		glog.Errorf("Failed to stat %s (%v)", devicePath, err)
+		return false
+	}
+	if !deviceDir.IsDir() {
+		return false
+	}
 
-	lines := strings.Split(output, "\n")
+	data, err := os.ReadFile(path.Join(devicePath, "device", "vendor"))
+	if errors.Is(err, os.ErrNotExist) {
+		// File doesn't exist but that is fine
+		return false
+	} else if err != nil {
+		glog.Errorf("Error reading %s (%v)",
+			path.Join(devicePath, "device", "vendor"), err)
+		return false
+	}
+
+	vendor := strings.TrimSuffix(string(data), "\n")
+	return vendor == solarflareVendor
+}
 
-	var interfaces []string
+// Get numa node from sysfs files.
+// -1 means no specific numa node / unknown
+func getNumaNode(devicePath string) int64 {
+	data, err := os.ReadFile(path.Join(devicePath, "device", "numa_node"))
+	if errors.Is(err, os.ErrNotExist) {
+		// File doesn't exist but that is fine, return -1
+		return -1
+	} else if err != nil {
+		glog.Errorf("Error reading %s (%v)",
+			path.Join(devicePath, "device", "vendor"), err)
+		return -1
+	}
+	numaString := strings.TrimSuffix(string(data), "\n")
+	node, err := strconv.ParseInt(numaString, 10, 64)
+	if err != nil {
+		glog.Errorf("Error parse int from string %s (%v)",
+			string(data), err)
```

Comment:
nit:

```suggestion
			numaString, err)
```

`string(data)` is also OK, but that's not what we give to `strconv.ParseInt()`.